server.go

```diff
@@ -10,11 +10,22 @@ import (
 	"google.golang.org/grpc"
 )
 
-// PreregisterServices takes a gRPC server and pre-initializes all counters to 0.
-// This allows for easier monitoring in Prometheus (no missing metrics), and should be called *after* all services have
-// been registered with the server.
-func Register(server *grpc.Server) {
-	serviceInfo := server.GetServiceInfo()
+// Server defines the interface that is required for grpc_prometheus to Register.
+type Server interface {
+	GetServiceInfo() map[string]grpc.ServiceInfo
+}
+
+// Register takes a gRPC server and pre-initializes all counters to 0.
+// This allows for easier monitoring in Prometheus (no missing metrics), and
+// should be called *after* all services have been registered with the server.
+func Register(server Server) {
```

Comment:
As long as `Register` accepts something that `*grpc.Server` implements, it is fine.